src/api/admin.py

```diff
@@ -0,0 +1,385 @@
+from src.infrastructure.config.app_config import STATIC_DIR
+from src.infrastructure.database.database_config import get_db
+from src.infrastructure.auth.auth_utils import admin_permission
+from src.domain.user import UserAdminInfo, PaginatedUserAdminInfo
+from src.domain.alcohol import AlcoholCreate, Alcohol, AlcoholUpdate
+from src.infrastructure.database.models.user import UserDatabaseHandler
+from src.infrastructure.database.models.alcohol import AlcoholDatabaseHandler
+from src.domain.reported_errors import ReportedError, PaginatedReportedErrorInfo
+from src.infrastructure.exceptions.users_exceptions import UserNotFoundException
+from src.infrastructure.exceptions.alcohol_exceptions import AlcoholExistsException
+from src.domain.alcohol_category import AlcoholCategoryDelete, AlcoholCategoryCreate
+from src.infrastructure.exceptions.validation_exceptions import ValidationErrorException
+from src.infrastructure.database.models.reported_error import ReportedErrorDatabaseHandler
+from src.infrastructure.database.models.alcohol_category import AlcoholCategoryDatabaseHandler
+from src.infrastructure.database.models.alcohol_category.mappers import map_to_alcohol_category
+from src.infrastructure.exceptions.alcohol_categories_exceptions import AlcoholCategoryExistsException
+from src.domain.alcohol_category import PaginatedAlcoholCategories, AlcoholCategory, AlcoholCategoryUpdate
+
+router = APIRouter(prefix='/admin', tags=['admin'], dependencies=[Depends(admin_permission)])
+
+
+@router.get(
+    path='/users',
+    response_model=PaginatedUserAdminInfo,
+    status_code=status.HTTP_200_OK,
+    summary='[For Admin] Read users',
+    response_model_by_alias=False
+)
+async def search_users(
+        limit: int = 10,
+        offset: int = 0,
+        username: str = '',
+        db: Database = Depends(get_db)
+):
+    """
+    Search for users with pagination by optional username
+    """
+    users = await UserDatabaseHandler.get_users(db.users, limit, offset, username)
+    total = await UserDatabaseHandler.count_users(db.users, username)
+    return PaginatedUserAdminInfo(
+        users=users,
+        page_info=PageInfo(
+            limit=limit,
+            offset=offset,
+            total=total
+        )
+    )
+
+
+@router.get(
+    path='/users/{user_id}',
+    response_model=UserAdminInfo,
+    status_code=status.HTTP_200_OK,
+    summary='[For Admin] Read user information',
+    response_model_by_alias=False,
+)
+async def get_user(
+        user_id: str,
+        db: Database = Depends(get_db)
+):
+    """
+    Read user information
+    """
+    db_user = await UserDatabaseHandler.get_user_by_id(db.users, user_id)
+    if not db_user:
+        raise UserNotFoundException()
+    return db_user
+
+
+@router.put(
+    path='/users/{user_id}',
+    status_code=status.HTTP_204_NO_CONTENT,
+    summary='[For Admin] Ban or unban user',
+)
+async def ban_user(
+        user_id: str,
+        to_ban: bool = True,
```

Comment:
Trochę dziwna nazwa, ale sam nie mam lepszego pomysłu

Reply:
może banned?

Reply:
odnośnie linijki 85

Reply:
Chyba nie, co sugeruje banned? `zbanowany`? 
A tak to masz `zbanować`/`do zbanowania`